src/main/java/com/example/everkan/email/EmailService.java

```diff
@@ -0,0 +1,44 @@
+package com.example.everkan.email;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Service
+public class EmailService implements EmailSender {
+
+    private final static Logger LOGGER = LoggerFactory
+            .getLogger(EmailService.class);
+    private final String from;
+    private final JavaMailSender mailSender;
+
+    public EmailService(@Value("${app.email_address}") String from, JavaMailSender mailSender) {
+        this.from = from;
+        this.mailSender = mailSender;
+    }
+
+    @Override
+    @Async
+    public void send(String to, String email) {
+        try {
+            MimeMessage mimeMessage = mailSender.createMimeMessage();
+            MimeMessageHelper helper =
+                    new MimeMessageHelper(mimeMessage, "utf-8");
+            helper.setText(email, true);
+            helper.setTo(to);
+            helper.setSubject("Confirm your email");
+            helper.setFrom(from);
+            mailSender.send(mimeMessage);
+        } catch (MessagingException e) {
+            LOGGER.error("failed to send email", e);
+            throw new IllegalStateException("failed to send email");
```

Comment:
You can pass the original exception as cause to the next one (`IllegalStateException`).